tests/test_uploads.py

```diff
@@ -0,0 +1,143 @@
+
+from webcompat import app
+
+
+class TestUploads(unittest.TestCase):
+    # modified from http://prschmid.blogspot.com/2013/05/unit-testing-flask-file-uploads-without.html
+    def setUp(self):
+        app.config['TESTING'] = True
+        self.app = app
+        self.test_client = self.app.test_client()
+
+    def tearDown(self):
+        pass
+
+    def testGet(self):
+        '''Test that /upload/ doesn't let you GET.'''
+        rv = self.test_client.get('/upload/')
+        self.assertEqual(rv.status_code, 404)
+
+    def testBadUploads(self):
+        # Loop over some files and the status codes that we are expecting
+        for filename, status_code in (
+                ('foo.xxx', 415),
+                ('foo', 415),
+                ('foo.rb', 415)):
+
+            # The reason why we are defining it in here and not outside
+            # this method is that we are setting the filename of the
+            # TestingFileStorage to be the one in the for loop. This way
+            # we can ensure that the filename that we are "uploading"
+            # is the same as the one being used by the application
+            class TestingRequest(Request):
+                """A testing request to use that will return a
+                TestingFileStorage to test the uploading."""
+                @property
+                def files(self):
+                    d = MultiDict()
+                    d['image'] = TestingFileStorage(filename=filename)
+                    return d
+
+            self.app.request_class = TestingRequest
+            test_client = self.app.test_client()
+
+            rv = test_client.post(
+                '/upload/',
+                data=dict(
+                    file=(StringIO('Fake image data'), filename),
+                ))
+            self.assertEqual(rv.status_code, status_code)
+
+    def testGoodUploads(self):
+        # Loop over some files and the URLs that we are expecting back
+        for filename, status_code in (
+                ('foo.png', 201),
+                ('foo.jpg', 201),
+                ('foo.gif', 201),
+                ('foo.jpe', 201),
+                ('foo.jpeg', 201),
+                ('foo.bmp', 201)):
+
+            class TestingRequest(Request):
+                """A testing request to use that will return a
+                TestingFileStorage to test the uploading."""
+                @property
+                def files(self):
+                    d = MultiDict()
+                    d['image'] = TestingFileStorage(filename=filename)
+                    return d
+
+            self.app.request_class = TestingRequest
+            test_client = self.app.test_client()
+
+            rv = test_client.post(
+                '/upload/',
+                data=dict(
+                    file=(StringIO('Fake image data'), filename),
+                ))
+            self.assertEqual(rv.status_code, status_code)
+
+
+class TestingFileStorage(FileStorage):
```

Comment:
On the other hand You do not necessary needs this :)

add  at the top

``` python
from flaskext.uploads import TestingFileStorage
```

And then when you need to use it with the modified content_type and the filename you just need to do

```python
d['image'] = TestingFileStorage(filename=filename, content_length=-1)
```
I tried and the tests are working.